File: backend/app/models/expense.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Float,
    Date,
    DateTime,
    ForeignKey
)
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

from app.database.session import Base


class Expense(Base):
    __tablename__ = "expenses"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    title = Column(
        String(255),
        nullable=False
    )

    amount = Column(
        Float,
        nullable=False
    )

    category = Column(
        String(100),
        nullable=False
    )

    expense_date = Column(
        Date,
        nullable=False
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False
    )

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )

    user = relationship(
        "User",
        back_populates="expenses"
    )

File: frontend/src/types/expense.ts
export interface Expense {
    id: number;
    title: string;
    amount: number;
    category: string;
    expense_date: string;
}

export interface CreateExpenseRequest {
    title: string;
    amount: number;
    category: string;
    expense_date: string;
}

export interface UpdateExpenseRequest {
    title: string;
    amount: number;
    category: string;
    expense_date: string;
}